R/nefsc_groundfish_access.R

```diff
@@ -955,9 +955,126 @@ add_area_stratification <- function(survdat_weights, include_epu = F, box_locati
+#'
+#' @param survdat_clean Survdat data, after usual preparations are completed.
+#' These include removal of old strata, labeling of areas of interest, and inclusion
+#' of the annual effort in each.
+#' @param box_location String indicating value to pass to `boxpath_switch`
+#'
+#' @return survdat trawl dataframe containing one row for every unique tow
+#' @export
+#'
+#' @examples
+#' # not run
+#' # get_survdat_tows(survdat_clean = survdat_clean)
+get_survdat_tows <- function(survdat_clean) {
+
+
+  #### 1. Filter SURVDAT to unique tows and keep columns of interest   ####
+
+  # Get unique tows
+  survdat_tows <- survdat_clean %>%
+    dplyr::distinct(id, est_towdate, est_year, est_month, est_day, season, svvessel, decdeg_beglat, decdeg_beglon, survey_area, avgdepth, surftemp, surfsalin, bottemp, botsalin) %>%
+    dplyr::filter(!is.na(decdeg_beglat) & !is.na(decdeg_beglon))
+
+  # Return it
+  return(survdat_tows)
+}
+
+###################################################### _
+
+# # Tests
+# survdat_tows <- get_survdat_tows(survdat_clean = survdat_clean)
+
+###################################################### __####
+
+###################################################### _
+#' @title make SURVDAT tidy occupancy dataset
+#'
+#' @description make a tidy occupancy dataset from the cleaned SURVDAT dataset, where each row represents a unique tow - species - occupancy record for every species in species_keep. In creating this occupancy dataset, we take the total biomass or abundance across sex/sizes and impute "absence" observations, such that there is a record for every species in `species_keep` at every unique tow.
+#'
+#' @param survdat_clean Survdat data, after usual preparations are completed.
+#' These include removal of old strata, labeling of areas of interest, and inclusion
+#' of the annual effort in each.
+#' @param species_keep Vector of character species names or vector of numeric species codes to filter SURVDAT['comname'] (if character), or SURVDAT['svspp'] (if numeric)
+#'
+#' @return tidy occupancy dataframe containing one row for every unique tow-species
+#' @export
+#'
+#' @examples
+#' # not run
+#' # make_survdat_occu(survdat_clean = survdat_clean, species_keep = c("atlantic cod", "offshore hake"))
+make_survdat_occu <- function(survdat_clean, species_keep){
+
+
+  #### 1. Filter SURVDAT to species of interest based on character or numeric vector `species_keep` and aggregate abundance and biomass data for these species at each tow location   ####
+  if (all(is.character(species_keep))) {
+    presence_data <- survdat_clean %>%
+      dplyr::filter(., comname %in% species_keep) %>%
+      dplyr::group_by(., id, svspp, comname) %>%
+      dplyr::summarise(
+        sum_abundance = sum(abundance),
+        sum_biomass_kg = ifelse(length(unique(biomass_kg == 1)), unique(biomass_kg), sum(biomass_kg))
```

Comment:
The sum(biomass_kg) part of the "else" statement will silently multiply biomass values here

Reply:
Definitely mucked up parentheses there, either way though like your suggestion for `sum(unique(biomass_kg))` to skip the `ifelse` entirely